src/window.rs

```diff
@@ -251,52 +262,62 @@ impl WindowWrapper {
-                        (physical_size.height as f32 * new_dpis.1 / self.previous_dpis.1) as u32,
-                    )
-                    .unwrap();
-                self.previous_dpis = new_dpis;
+        if let Some(mut window) = self.window.take() {
+            if let Ok(new_size) = LogicalSize::new(&window) {
+                if self.previous_size != new_size {
+                    handle_new_grid_size(new_size, &self.renderer);
+                    self.previous_size = new_size;
+                }
+            }
+
+            if let Ok(new_dpis) = dpis(&window) {
+                if self.previous_dpis != new_dpis && cfg!(target_os = "windows") {
+                    let physical_size = PhysicalSize::new(&window);
+                    window
+                        .set_size(
+                            (physical_size.width as f32 * new_dpis.0 / self.previous_dpis.0) as u32,
+                            (physical_size.height as f32 * new_dpis.1 / self.previous_dpis.1)
+                                as u32,
+                        )
+                        .unwrap();
+                    self.previous_dpis = new_dpis;
+                }
             }
-        }
 
-        debug!("Render Triggered");
-        let current_size = self.previous_size;
-        if REDRAW_SCHEDULER.should_draw() || SETTINGS.get::<WindowSettings>().no_idle {
-            let renderer = &mut self.renderer;
-            if self
-                .skulpin_renderer
-                .draw(&self.window, |canvas, coordinate_system_helper| {
-                    let dt = 1.0 / (SETTINGS.get::<WindowSettings>().refresh_rate as f32);
-
-                    if renderer.draw(canvas, coordinate_system_helper, dt) {
-                        handle_new_grid_size(current_size, &renderer)
-                    }
-                })
-                .is_err()
-            {
-                error!("Render failed. Closing");
-                return false;
+            debug!("Render Triggered");
+            let current_size = self.previous_size;
+            if REDRAW_SCHEDULER.should_draw() || SETTINGS.get::<WindowSettings>().no_idle {
+                let renderer = &mut self.renderer;
+                if self
+                    .skulpin_renderer
+                    .draw(&window, |canvas, coordinate_system_helper| {
+                        let dt = 1.0 / (SETTINGS.get::<WindowSettings>().refresh_rate as f32);
+
+                        if renderer.draw(canvas, coordinate_system_helper, dt) {
+                            handle_new_grid_size(current_size, &renderer)
+                        }
+                    })
+                    .is_err()
+                {
+                    error!("Render failed. Closing");
+                    return false;
+                }
             }
+            self.window = Some(window);
+            return true;
         }
-        return true;
+        return false;
```

Comment:
Are you sure false is the correct value here? The render didn't fail, it just didn't draw. Same thing as when no redraw is scheduled. As is this will stop the draw loop. Is that the correct response?

Reply:
You're right. Just checked the logic with ui loop.